File: src/test/java/no/nav/fo/veilarbregistrering/bruker/FoedselsdatoTest.java
package no.nav.fo.veilarbregistrering.bruker;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.assertj.core.api.Assertions.assertThat;

public class FoedselsdatoTest {

    @Test
    public void alder_oker_ikke_for_foedselsdag() {
        Foedselsdato foedselsdato = new Foedselsdato(LocalDate.of(1978, 6, 23)) {

            @Override
            protected LocalDate dagensDato() {
                return LocalDate.of(2020, 2, 20);
            }
        };

        assertThat(foedselsdato.alder()).isEqualTo(41);
    }

    @Test
    public void foedselsdag_oker_alder() {
        Foedselsdato foedselsdato = new Foedselsdato(LocalDate.of(1978, 6, 23)) {

            @Override
            protected LocalDate dagensDato() {
                return LocalDate.of(2020, 6, 23);
            }
        };

        assertThat(foedselsdato.alder()).isEqualTo(42);
    }
}

File: src/test/java/no/nav/fo/veilarbregistrering/bruker/FoedselsdatoTest.kt
package no.nav.fo.veilarbregistrering.bruker

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.LocalDate

class FoedselsdatoTest {

    @Test
    fun `alder øker ikke for fødselsdag`() {
        val foedselsdato = Foedselsdato(LocalDate.of(1978, 6, 23))

        assertThat(foedselsdato.alderPaa(LocalDate.of(2020, 2, 20))).isEqualTo(41)
    }

    @Test
    fun `fødselsdag øker alder`() {
        val foedselsdato = Foedselsdato(LocalDate.of(1978, 6, 23))

        assertThat(foedselsdato.alderPaa(LocalDate.of(2020, 6, 23))).isEqualTo(42)
    }
}
